packages/flutter/lib/src/widgets/text_selection.dart

```diff
@@ -541,7 +541,13 @@ class TextSelectionOverlay {
   ///
   /// This is intended to be called when the [renderObject] may have changed its
   /// text metrics (e.g. because the text was scrolled).
-  void updateForScroll() => _updateSelectionOverlay();
+  void updateForScroll() {
+    _updateSelectionOverlay();
+    // This method may be called due to windows metrics changes. In that case,
+    // non of the properties in _selectionOverlay will change, but a rebuild is
+    // still needed.
+    _selectionOverlay.markNeedsBuild();
```

Comment:
Another bug i found after I fix the TextSelectionPoint == operator. This method will be called when windows insets change, previous it will always mark needs build because the selectionEndPoint will always think it received a different endpoints.